src/libsyntax/ast.rs

```diff
@@ -1940,18 +1949,24 @@ pub enum CrateSugar {
 
 pub type Visibility = Spanned<VisibilityKind>;
 
-#[derive(Clone, PartialEq, Eq, RustcEncodable, RustcDecodable, Hash, Debug)]
+#[derive(Clone, RustcEncodable, RustcDecodable, Debug)]
 pub enum VisibilityKind {
     Public,
     Crate(CrateSugar),
     Restricted { path: P<Path>, id: NodeId },
     Inherited,
 }
 
+impl VisibilityKind {
+    pub fn is_public(&self) -> bool {
```

Comment:
FIXME: Rename to `is_pub` for compatibility with https://github.com/rust-lang/rust/pull/51866.